### ISSUE TITLE
Hang on TT-RSS v18.8 (79c5035)

### ISSUE THREAD

**Issue by TsarFox** [fasheng/elfeed-protocol#13]
Hi, I'm trying to connect to a TT-RSS instance (v18.8 (79c5035)), but whenever I update, Elfeed gets stuck at `0 jobs pending, 1 active...` and `Error running timer ‘elfeed-curl--call-callback’: (void-function string-empty-p)` appears in `*Messages*`.

Relevant contents of `init.el`:
```
(require 'elfeed-protocol)
(setq elfeed-feeds (list "ttrss+https://jakob:[REDACTED]@rss.tildeverse.org"))
(elfeed-protocol-enable)
```

Contents of `*elfeed-log*`:
```
[2019-01-09 11:16:19] [info]: Elfeed update: January  9 2019 11:16:19 EST
[2019-01-09 11:16:19] [debug]: elfeed-protocol-ttrss: check is logged in
[2019-01-09 11:16:20] [debug]: retrieve (https://rss.tildeverse.org/api/)
[2019-01-09 11:16:20] [debug]: elfeed-protocol-ttrss: update feed list
[2019-01-09 11:16:20] [debug]: retrieve (https://rss.tildeverse.org/api/)
[2019-01-09 11:16:21] [debug]: elfeed-protocol-ttrss: update entries with action init, arg nil
[2019-01-09 11:16:21] [debug]: retrieve (https://rss.tildeverse.org/api/)
[2019-01-09 11:16:22] [debug]: elfeed-protocol-ttrss: parsing entries, first-entry-id: -1 last-entry-id: -1
[2019-01-09 11:16:22] [debug]: elfeed-protocol-ttrss: parsed 0 entries with 0.000035s, first-entry-id: -1 last-entry-id: -1
[2019-01-09 11:16:22] [debug]: retrieve (https://rss.tildeverse.org/api/)
[2019-01-09 11:16:23] [debug]: elfeed-protocol-ttrss: parsing entries, first-entry-id: -1 last-entry-id: -1
```

**Comment by fasheng** [comment]
TT-RSS `v17.x` test passed before, I will try to find a docker image for `v18.8` to reproduce your issue later.

**Comment by fasheng** [comment]
Hi, I didn't reproduce your issue finally. Here is my environment:
- emacs `26.1`
- elfeed `3.1.0`
- elfeed-protocol `0.5.7`
- TT-RSS `v18.12` (docker image `linuxserver/tt-rss:154` which only owns the default "Tiny Tiny RSS: Forum" feed)
- emacs full config
  ```lisp
  (setq elfeed-log-level 'debug)
  (setq elfeed-use-curl t)
  (elfeed-set-timeout 36000)
  (setq elfeed-feeds (list "ttrss+http://admin:password@localhost"))
  (elfeed-protocol-enable)
  ```
And the full log for first update operation:
```
[2019-01-11 10:58:43] [info]: Elfeed update: January 11 2019 10:58:43 CST
[2019-01-11 10:58:43] [debug]: elfeed-protocol-ttrss: login
[2019-01-11 10:58:43] [debug]: retrieve (http://localhost/api/)
[2019-01-11 10:58:43] [debug]: elfeed-protocol-ttrss: update feed list
[2019-01-11 10:58:43] [debug]: retrieve (http://localhost/api/)
[2019-01-11 10:58:48] [debug]: elfeed-protocol-ttrss: update entries with action init, arg nil
[2019-01-11 10:58:48] [debug]: retrieve (http://localhost/api/)
[2019-01-11 10:58:48] [debug]: elfeed-protocol-ttrss: parsing entries, first-entry-id: -1 last-entry-id: -1
[2019-01-11 10:58:48] [debug]: elfeed-protocol-ttrss: parsed 0 entries with 0.000082s, first-entry-id: -1 last-entry-id: -1
[2019-01-11 10:58:48] [debug]: retrieve (http://localhost/api/)
[2019-01-11 10:58:49] [debug]: elfeed-protocol-ttrss: parsing entries, first-entry-id: -1 last-entry-id: -1
[2019-01-11 10:58:49] [debug]: elfeed-protocol-ttrss: parsed 10 entries with 0.005805s, first-entry-id: 1 last-entry-id: 11
[2019-01-11 10:58:49] [debug]: retrieve (http://localhost/api/)
[2019-01-11 10:58:49] [debug]: elfeed-protocol-ttrss: parsing entries, first-entry-id: 1 last-entry-id: 1
[2019-01-11 10:58:49] [debug]: elfeed-protocol-ttrss: parsed 10 entries with 0.003669s, first-entry-id: 1 last-entry-id: 11
```

You could have a try again with my environment to see if it works.

**Comment by TsarFox** [comment]
So I tried again with `(setq elfeed-use-curl nil)` and realized that I totally misinterpreted the error in my report. Really, the answer was staring me right in the eyes: `error in process filter: Symbol’s function definition is void: string-empty-p`. `string-empty-p` wasn't defined. So I defined a `string-empty-p` and suddenly everything started working.

A little bit of digging reveals that `string-empty-p` is provided by `subr-x`, which wasn't loaded. I added the require in #14.

**Comment by TsarFox** [comment]
So I tried again with `(setq elfeed-use-curl nil)` and realized that I totally misinterpreted the error in my report. Really, the answer was staring me right in the eyes: `error in process filter: Symbol’s function definition is void: string-empty-p`. `string-empty-p` wasn't defined. So I defined a `string-empty-p` and suddenly everything started working.

A little bit of digging reveals that `string-empty-p` is provided by `subr-x`, which wasn't loaded. I added the require in #14.
